packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -127,12 +135,21 @@ export class Relayer {
   // Deposit params are incorrect, then the BridgePool's computed deposit hash will be different and the relay won't be
   // found. So, assuming that the relay contains a matching deposit hash, this bot's job is to only consider speeding up
   // relays that are valid, otherwise the bot might lose money without recourse on the relay.
-  private async isRelayValid(relay: Relay, deposit: Deposit): Promise<boolean> {
-    return (
-      relay.realizedLpFeePct.toString() ===
-        (await this.l1Client.calculateRealizedLpFeePctForDeposit(deposit)).toString() &&
-      relay.depositContract === deposit.depositContract
-    );
+  private async isRelayValid(relay: Relay, deposit: Deposit): Promise<{ isValid: boolean; reason: string }> {
+    const relayRealizedLpFeePct = relay.realizedLpFeePct.toString();
+    const expectedRelayRealizedLpFeePct = (await this.l1Client.calculateRealizedLpFeePctForDeposit(deposit)).toString();
+    if (relayRealizedLpFeePct !== expectedRelayRealizedLpFeePct)
+      return {
+        isValid: false,
+        reason: `relayRealizedLpFeePct: ${relayRealizedLpFeePct}!=expectedRelayRealizedLpFeePct: ${expectedRelayRealizedLpFeePct}`,
+      };
+    if (relay.depositContract !== deposit.depositContract)
```

Comment:
nit: why not use `else if`?

Reply:
Also this test case seems really hard to recreate, I'm curious if you tested it?